ray-operator/controllers/common/pod.go

```diff
@@ -90,7 +90,18 @@ func DefaultWorkerPodTemplate(instance rayiov1alpha1.RayCluster, workerSpec rayi
 		Name:          "metrics",
 		ContainerPort: int32(DefaultMetricsPort),
 	}
-	podTemplate.Spec.Containers[0].Ports = append(podTemplate.Spec.Containers[0].Ports, metricsPort)
+	dupIndex := -1
```

Comment:
@Jeffwan This is a bug fix that will create multiple container ports with the same name error.

Reply:
@scarlet25151

Reply:
How can we reproduce the issue? I think the error only happens when we already have a metrics port define in template?

Reply:
yes, it's my bad, in the past the yaml file would not include metrics port and I forgot the duplicate check here